### ISSUE TITLE
关于huggingface的finetune前的结果出入

### ISSUE THREAD

**Issue by Mrwangkedong** [thu-coai/EVA#70]
我使用的是[huginface ]的模型。

对于base model，finetune kdconv数据集的film domin，得到的valid 结果是：
dist-1: 1.9876 |  dist-2: 15.466 |  dist-3: 30.441 |  dist-4: 43.694 | 
bleu-1: 29.431 |  bleu-2: 19.297 |  bleu-3: 13.997 |  bleu-4: 10.812 | 
f1: 30.208 |  rouge-l: 26.64 |  rouge-1: 76.119 |  rouge-2: 40.532 |  loss: 0.46232 | 

【暂时没有finetune完，稍后补充】对于large model，finetune kdconv数据集的film domin，得到的valid 结果是：
dist-1: 0.21527 |  dist-2: 0.50083 |  dist-3: 0.73138 |  dist-4: 0.96529 | 
bleu-1: 17.382 |  bleu-2: 8.338 |  bleu-3: 4.8573 |  bleu-4: 3.0812 | 
f1: 19.121 |  rouge-l: 15.913 |  rouge-1: 11.394 |  rouge-2: 2.5146 |  loss: 0.42247 |

lr=0.0002  生成参数do-sample: False  max-generation-length: 30  min-generation-length: 2  num-beams: 4  no-repeat-ngram-size: 4  repetition-penalty: 1.6  length-penalty: 1.6

**Comment by t1101675** [comment]
您好，您的问题是什么？

**Comment by t1101675** [comment]
可以检查一下解码是否正确，一般 loss 相似的情况下生成结果指标不会差距这么大

**Comment by Mrwangkedong** [comment]
应该不会，我使用提供的huginface 分支提供的eva_evaluate.py进行验证。对于仓库里给出的处理好的kdconv，在finetune前的结果如下：
base finetune前
dist-3: 27.219 |  dist-4: 40.535 |  bleu-4: 0.73811 |  f1: 17.952 |  rouge-l: 14.187 |  rouge-1: 52.013 |  rouge-2: 17.117 |  loss: 4.0454 |

large finetune前
dist-3: 21.305 |  dist-4: 31.217 |  bleu-4: 0.40356 |  f1: 14.908 |  rouge-l: 11.521 |  rouge-1: 44.481 |  rouge-2: 12.149 |  loss: 1.8331 |

xlarge finetune前
dist-3: 34.728 |  dist-4: 50.451 |  bleu-4: 1.0312 |  f1: 17.961 |  rouge-l: 14.394 |  rouge-1: 59.829 |  rouge-2: 21.11 |  loss: 2.3548 |

**Comment by Mrwangkedong** [comment]
我的两个finetune是使用相同的解码策略

**Comment by xwwwwww** [comment]
测一下fine-tune前的zero-shot指标呢？

**Comment by Mrwangkedong** [comment]
> 测一下fine-tune前的zero-shot指标呢？
我是使用huggingface分支下面的src/scripts/eva_inference_static.sh进行验证的。
base：
Eval result:  dist-3: 26.785 |  dist-4: 39.631 |  bleu-4: 0.77208 |  f1: 17.959 |  rouge-l: 14.098 |  rouge-1: 52.961 |  rouge-2: 17.4 |  loss: 6.5944 |
large:
dist-3: 32.679 |  dist-4: 45.19 |  bleu-4: 0.41946 |  f1: 15.185 |  rouge-l: 11.86 |  rouge-1: 41.244 |  rouge-2: 11.733 |  loss: 1.3344 |

**Comment by t1101675** [comment]
我们测了一下，zero-shot large 模型在 kdconv 上 zero-shot 的结果，生成的指标上确实不如 base，只能说可能不同 domain 模型的相对表现会有一定差异。
Large:
Eval result:  dist-3: 46.973 |  dist-4: 59.176 |  bleu-4: 0.52577 |  f1: 15.643 |  rouge-l: 12.004 |  rouge-1: 35.87 |  rouge-2: 11.485 |  loss: 1.78 | 
Base: 
Eval result:  dist-3: 53.671 |  dist-4: 68.908 |  bleu-4: 0.90753 |  f1: 18.728 |  rouge-l: 14.485 |  rouge-1: 40.316 |  rouge-2: 15.105 |  loss: 4.0595 | 

我们也跑了一下 fine-tune 的结果
Large: 
Eval result:  dist-3: 59.124 |  dist-4: 78.833 |  bleu-4: 3.2913 |  f1: 20.124 |  rouge-l: 18.554 |  rouge-1: 79.135 |  rouge-2: 35.740 |  loss: 2.3963 |
Base:
Eval result:  dist-3: 58.491 |  dist-4: 77.947 |  bleu-4: 3.231 |  f1: 19.771 |  rouge-l: 17.029 |  rouge-1: 78.059 |  rouge-2: 35.738 |  loss: 2.5184 |

fine-tune 后还是 large 模型相对更好一些

**Comment by t1101675** [comment]
> 我使用的是[huginface ]的模型。
> 
> 对于base model，finetune kdconv数据集的film domin，得到的valid 结果是： dist-1: 1.9876 | dist-2: 15.466 | dist-3: 30.441 | dist-4: 43.694 | bleu-1: 29.431 | bleu-2: 19.297 | bleu-3: 13.997 | bleu-4: 10.812 | f1: 30.208 | rouge-l: 26.64 | rouge-1: 76.119 | rouge-2: 40.532 | loss: 0.46232 |
> 
> 【暂时没有finetune完，稍后补充】对于large model，finetune kdconv数据集的film domin，得到的valid 结果是： dist-1: 0.21527 | dist-2: 0.50083 | dist-3: 0.73138 | dist-4: 0.96529 | bleu-1: 17.382 | bleu-2: 8.338 | bleu-3: 4.8573 | bleu-4: 3.0812 | f1: 19.121 | rouge-l: 15.913 | rouge-1: 11.394 | rouge-2: 2.5146 | loss: 0.42247 |
> 
> lr=0.0002 生成参数do-sample: False max-generation-length: 30 min-generation-length: 2 num-beams: 4 no-repeat-ngram-size: 4 repetition-penalty: 1.6 length-penalty: 1.6

这里 base model 的结果显然过高，不太正常，可以检查下是不是 fine-tune 有问题

**Comment by Mrwangkedong** [comment]
> 3.2913 | f1: 20.124 | rouge-l: 18.554 | rouge-1: 79.135 | rouge-2: 35.740 | loss: 2.3963 |
> Base:

好的，请问后续可以更新一下finetune代码吗？参考一下

**Comment by t1101675** [comment]
这个结果是拿现在仓库里的 fine-tune 代码跑出来的，如果要 huggingface 版本的话可能需要等一段时间

**Comment by Mrwangkedong** [comment]
> 

好的，有劳了

**Comment by zzzzhaozhao** [comment]
@Mrwangkedong 请问您基于huggingface的finetune是否可以开源，初学者想要参考一下，如有打扰请见谅，万分感谢！

**Comment by Mrwangkedong** [comment]
我的代码可能有问题，在issue中和作者的对话你应该可以看到。请问还需要吗？需要的话过几天我整理一下发给你，顺便一起找找错误~&nbsp;


www.1241925557
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年12月8日(星期四) 中午11:48
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [thu-coai/EVA] 关于huggingface的finetune前的结果出入 (Issue #70)





 
@Mrwangkedong 请问您基于huggingface的finetune是否可以开源，初学者想要参考一下，如有打扰请见谅，万分感谢！

**Comment by zzzzhaozhao** [comment]
如果可以的话，那真是太好了！您有时间可以发我一份，非常感谢！

**Comment by Mrwangkedong** [comment]
> 如果可以的话，那真是太好了！您有时间可以发我一份，非常感谢！

[https://github.com/Mrwangkedong/EVA2.0_fintune](url)
